features/attached_commands.feature

```diff
@@ -543,6 +546,7 @@ Feature: Command behaviour when attached to an UA subscription
         And I run `python3 /usr/lib/ubuntu-advantage/timer.py` with sudo
         Then I verify that running `grep -q activityToken /var/lib/ubuntu-advantage/private/machine-token.json` `with sudo` exits `0`
         And I verify that running `grep -q activityID /var/lib/ubuntu-advantage/private/machine-token.json` `with sudo` exits `0`
+        And I verify that running `grep -q activityPingInterval /var/lib/ubuntu-advantage/private/machine-token.json` `with sudo` exits `0`
```

Comment:
Should we be calling the equivalent of `ua config set metering_interval <activityPingInterval>` with this value in `report_machine_activity`?

Reply:
I think we decided we don't want to override user-customized values in `/etc/ubuntu-advantage/uaclient.conf` as some customers will choose to completely disable this service or throttle the frequency lower than the contract sets.  But, per my comment above, I think our TimedJob.run_interval_seconds should be aware of this valiue for the metering job somehow as we want to prefer the contract server's recommendation of longer  in this scenario:
```python
if uaclient.conf.metering_timer == 0:
   return uaclient.conf.metering_timer
return max(activityPingInteuaclient.conf.metering_timerrval, uaclient.conf.metering_timer)
```

Reply:
Very good point. I'm wondering how we can avoid limiting ourselves in the future though. 

Because we are including 4 hours in the default conf, this snippet would effectively mean we could never reduce the pingInterval below 4 hours via the contract server. Maybe thats okay though?

Reply:
I think the subclass of TimedJob, overwriting the run_interval_seconds method, is good for this scenario.
I also remember some discussion we had about this situation, where the contract server would put a lower expected interval, and from our side (or for the workflow as a whole) would not be a problem, but this is worth a check with CS I guess.

Reply:
Yes, I think that's the right approach here